lib/membrane_mp4/demuxer/isom.ex

```diff
@@ -141,12 +142,15 @@ defmodule Membrane.MP4.Demuxer.ISOM do
           fsm_state: :samples_info_present_and_all_pads_connected
         } = state
       ) do
-    {samples, rest, samples_info} =
-      SamplesInfo.get_samples(state.samples_info, state.partial <> buffer.payload)
+    {samples, {rest, rest_iterator}, samples_info} =
```

Comment:
The `rest_iterator` could be initialized in `SamplesInfo.get_samples_info` and kept in `samples_info`

Reply:
done